extensions/amp-form/0.1/amp-form.js

```diff
@@ -769,12 +769,20 @@ export class AmpForm {
   }
 
   /**
-   * Transition the form to the submit success state.
+   * Transition the form to the submit-success or submit-error state depending on the response status.
    * @param {!JsonObject} response
    * @return {!Promise}
    * @private
    */
-  handleSsrTemplateSuccess_(response) {
+  handleSsrTemplateResponse_(response) {
+    if (
+      response['init'] &&
+      response['init']['status'] &&
+      response['init']['status'] >= 300
```

Comment:
Nit: Checking truthiness of `response['init']['status']` is redundant. We can also avoid a few repeated strings:

```js
const init = response['init'];
if (init) {
  const status = init['status'];
  if (status >= 300) {
    return this.handleSubmitFailure_(status, response);
  }
}
```